bundles/org.openhab.binding.gridbox/src/main/java/org/openhab/binding/gridbox/internal/api/GridBoxApi.java

```diff
@@ -0,0 +1,277 @@
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.gridbox.internal.api;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.gridbox.internal.GridBoxConfiguration;
+import org.openhab.binding.gridbox.internal.model.LiveData;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+/**
+ * The {@link GridBoxApi} is responsible for executing the HTTP calls to the GridBox API. Calls are executed
+ * synchronously, so the functions should be called from a parallel thread.
+ *
+ * @author Benedikt Kuntz - Initial contribution
+ */
+@NonNullByDefault
+public class GridBoxApi {
+
+    public class GridBoxApiSystemNotFoundException extends Exception {
+
+        private static final long serialVersionUID = 2485670225601258718L;
+
+        public GridBoxApiSystemNotFoundException(String message) {
+            super(message);
+        }
+    }
+
+    public class GridBoxApiException extends Exception {
+
+        private static final long serialVersionUID = -5295192044532589122L;
+
+        public GridBoxApiException(String message) {
+            super(message);
+        }
+    }
+
+    public class GridBoxApiAutheticationException extends Exception {
+
+        private static final long serialVersionUID = 7626147923372849513L;
+
+        public GridBoxApiAutheticationException(String message) {
+            super(message);
+        }
+    }
+
+    private static final HttpClient HTTP_CLIENT = HttpClient.newHttpClient();
+    private static final Duration TIMEOUT_DURATION = Duration.ofSeconds(5);
+    private static final Gson GSON = new Gson();
+
+    private final Logger logger = LoggerFactory.getLogger(GridBoxApi.class);
+
+    /**
+     * Fetch the ID Token required for the GridBox API calls by querying an OAuth request.
+     *
+     * @param config The {@link GridBoxConfiguration} to use, must contain valid user email/password
+     * @return The fetched ID Token. Will always be non-null, if no exception occurs.
+     * @throws IOException see {@link HttpClient#send(HttpRequest, java.net.http.HttpResponse.BodyHandler)}
+     * @throws InterruptedException see {@link HttpClient#send(HttpRequest, java.net.http.HttpResponse.BodyHandler)}
+     * @throws GridBoxApiAutheticationException Thrown in case of an invalid response of the authorization query, either
```

Comment:
```suggestion
     * @throws GridBoxApiAuthenticationException Thrown in case of an invalid response of the authorization query, either
```

Reply:
Good spot, thanks! The actual class name also had the typo in it...